src/RecordComponent.cpp

```diff
@@ -186,7 +186,12 @@ RecordComponent::flush(std::string const& name)
 void
 RecordComponent::read()
 {
+    if ( *hasBeenRead )
```

Comment:
No, this is just a regular flag. Just that the flag lives behind a shared pointer, since `RecordComponent` is a handle and the flag should be shared between instances.